Sources/SwiftPlot/Histogram.swift

```diff
@@ -276,111 +276,79 @@ extension Histogram: HasGraphLayout {
-                    let nextTopLeft = Point(topRight.x, histogramSeries.scaledBinFrequency[i+1])
-                    renderer.drawLine(startPoint: topRight,
-                                      endPoint: nextTopLeft,
-                                      strokeWidth: strokeWidth,
-                                      strokeColor: histogramSeries.color,
-                                      isDashed: false)
+            let xStart = Float(xMargin)
+            let xValues = stride(from: xStart, through: xStart + Float(binCount) * barWidth, by: barWidth)
+            
+            // One heights array for each series
+            var seriesHeights: [[Float]] = [[Float](repeating: 0.0, count: binCount + 2)]
+            
+            // Update `currentHeights` with the height from the series and add ìt to `heights`
+            var currentHeights = seriesHeights[seriesHeights.startIndex]
+            for series in allSeries {
+                for (newHeight, index) in zip(series.scaledBinFrequency, currentHeights.indices.dropFirst().dropLast()) {
+                    currentHeights[index] += newHeight
                 }
-                for series in histogramStackSeries {
-                    topLeft = Point(topLeft.x, currentHeight + series.scaledBinFrequency[i])
-                    topRight = Point(topRight.x, currentHeight + series.scaledBinFrequency[i])
-                    if (series.scaledBinFrequency[i] > 0) {
-                        renderer.drawLine(startPoint: topLeft,
-                                          endPoint: topRight,
-                                          strokeWidth: strokeWidth,
-                                          strokeColor: series.color,
-                                          isDashed: false)
-                        if (i != histogramSeries.bins-1) {
-                            var nextHeight = histogramSeries.scaledBinFrequency[i+1]
-                            for k in histogramStackSeries {
-                                nextHeight += k.scaledBinFrequency[i+1]
-                            }
-                            let nextTopLeft = Point(topRight.x, nextHeight)
-                            renderer.drawLine(startPoint: topRight,
-                                              endPoint: nextTopLeft,
-                                              strokeWidth: strokeWidth,
-                                              strokeColor: series.color,
-                                              isDashed: false)
-                        }
+                seriesHeights.append(currentHeights)
+            }
+            
+            // Draw only the line segments that will actually be visible, unobstructed from other lines that will be on top
+            // We iterate over the series in reverse to draw them from back to front
+            var seriesHeightsSlice = seriesHeights.reversed()[...]
+            var heightsS1Slice = seriesHeightsSlice.removeFirst()[...]
```

Comment:
I find that the names `Series1` or `S1` could be clearer. Maybe "foreground" and "background"?

Reply:
Thank you so much, I actually spent a lot of time trying to get the names clear but still manageably concise and still wasn't happy, but you just nailed it!